frontend/public/components/catalog-items.jsx

```diff
@@ -169,26 +209,104 @@ export class CatalogTileViewPage extends React.Component {
+    this.setState({ filters });
+  }
+
+  onFilterChange(filterType, id, value) {
+    const filters = { ...this.state.filters };
+    if (filterType === 'byName') {
+      filters[filterType].active = _.size(value) ? true : false;
+      filters[filterType].value = value;
+    } else {
+      filters[filterType][id].active = value;
+    }
+    this.setState({ filters });
+  }
+
   render() {
-    const { activeTabs, showAllItemsForCategory, currentCategories, numItems } = this.state;
+    const { activeTabs, showAllItemsForCategory, currentCategories, numItems, filters } = this.state;
+    const { clusterServiceClass, imageStream } = filters.byType;
     const activeCategory = showAllItemsForCategory ? _.find(currentCategories, { id: showAllItemsForCategory }) : null;
     const heading = activeCategory ? activeCategory.label : this.getCategoryLabel(_.first(activeTabs));
 
     return (
       <div className="co-catalog-page">
         <div className="co-catalog-page__tabs">
           { this.renderCategoryTabs() }
+          <FilterSidePanel id="filter-panel">
+            <FilterSidePanel.Category>
+              <FormControl type="text" placeholder="Filter by name..." bsClass="form-control"
+                value={filters.byName.value}
+                onChange={e => this.onFilterChange('byName', null, e.target.value)}
+              />
+            </FilterSidePanel.Category>
+            <FilterSidePanel.Category title="Type">
+              <FilterSidePanel.CategoryItem
+                count={clusterServiceClass.count}
+                checked={clusterServiceClass.active}
+                onChange={e => this.onFilterChange('byType', 'clusterServiceClass', e.target.checked)}
+              >
+                {clusterServiceClass.label}
+              </FilterSidePanel.CategoryItem>
+              <FilterSidePanel.CategoryItem
+                count={imageStream.count}
+                checked={imageStream.active}
+                onChange={e => this.onFilterChange('byType', 'imageStream', e.target.checked)}
+              >
+                {imageStream.label}
+              </FilterSidePanel.CategoryItem>
+            </FilterSidePanel.Category>
+          </FilterSidePanel>
         </div>
         <div className="co-catalog-page__content">
           <div>
             <div className="co-catalog-page__heading">{heading}</div>
             <div className="co-catalog-page__num-items">{numItems} items</div>
           </div>
-          <CatalogTileView>
+          {numItems > 0 && <CatalogTileView>
             {activeCategory
               ? this.renderCategoryTiles(activeCategory)
               : _.map(currentCategories, category => (category.numItems && category.id !== 'all') ? this.renderCategoryTiles(category) : null)}
-          </CatalogTileView>
+          </CatalogTileView>}
+          {numItems === 0 && <EmptyState>
+            <EmptyState.Title>No Results Match the Filter Criteria</EmptyState.Title>
```

Comment:
I find it much easier to read to do:
```
          {numItems === 0 && (
            <EmptyState>
              <EmptyState.Title>No Results Match the Filter Criteria</EmptyState.Title>
              ...
            </EmptyState>
          )}
  ```

Reply:
umm...that's not the norm in this particular repo. we usually do `{<cond> && <element...}`

Reply:
well, that's icky 😉